Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System;
+
+namespace BootCamp.Chapter
+{
+    public class BalanceStats
+    {
+        private readonly string _currency;
+        private readonly Account[] _accounts;
+        private const int _arrayBreak = 2;
+
+        public BalanceStats(string[] peopleAndBalances)
+        {
+            _currency = Settings.currency;
+            _accounts = AccountOps.BuildAccountList(peopleAndBalances);
+        }
+
+        /// <summary>
+        /// Return name and balance(current) of person who had the biggest historic balance.
+        /// </summary>
+        public string FindHighestBalanceEver()
+        {
+            if (_accounts is null)
+            {
+                return Messages.InvalidMessage;
+            }
+
+            Account highestBalanceAccount;
+            try
+            {
+                highestBalanceAccount = _accounts[0];
+                for (int i = 1; i < _accounts.Length; i++)
+                {
+                    if (_accounts[i].GetHighestBalance() > highestBalanceAccount.GetHighestBalance())
+                    {
+                        highestBalanceAccount = _accounts[i];
+                    }
+                }
+            }
+            catch (IndexOutOfRangeException)
+            {
+                return Messages.InvalidMessage;
+            }
+
+            return _accounts.Length > _arrayBreak && AccountOps.AreBalancesEqual(_accounts)
```

Comment:
this needs to be refactored in a method like BuildResult and use it for all statistics methods.

Reply:
statistics methods have different tests, I don't see a proper way to implement this.